protos/component_metadata/component_metadata.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+package mavsdk.rpc.component_metadata;
+
+import "mavsdk_options.proto";
+
+option java_package = "io.mavsdk.component_metadata";
+option java_outer_classname = "ComponentMetadataProto";
+
+// Access component metadata json definitions, such as parameters.
+service ComponentMetadataService {
+
+    /*
+     * Request metadata from a specific component
+     */
+    rpc RequestComponent(RequestComponentRequest) returns(EmptyResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    /*
+     * Request metadata from the autopilot component
+     */
+    rpc RequestAutopilotComponent(EmptyRequest) returns(EmptyResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    /*
+     * Register a callback that gets called when metadata is available
+     */
+    rpc SubscribeMetadataAvailable(MetadataAvailableRequest) returns(stream MetadataUpdateResponse) { option (mavsdk.options.async_type) = ASYNC; }
```

Comment:
Subscriptions are meant to be for things that get updated over time, rather than one time request/response. What is this one meant to be?

Reply:
Yes I meant it to be a subscription:
- each json type is returned immediately when available
- the same type could change over time (e.g. when rebooting the FC after an update, or dynamically, which is not implemented, but this API makes it possible)

Reply:
Right, that's true. In this case it should actually just be the subscription, without the request. So you subscribe and then you get updates as they come in.